### ISSUE TITLE
Inference: how to get K most-similar results? 

### ISSUE THREAD

**Issue by btseytlin** [KevinMusgrave/pytorch-metric-learning#132]
Hello!
Thanks for this amazing toolset. The code is great and it's really helping me.

After training a model I want to retrieve top-k most similar results, or k nearest neighbours, of a query image.

How do I do that? I tried the inference collab, but `InferenceModel` only provides interfaces to get matches within a batch.

The current way I am seeing is this:
1. Use the trained trunk + embedder to obtain embeddings of the whole training set
2. Use something like faiss to build an index
3. Obtain the embedding for query and use the faiss index to search for approximate NN

Is there an easier way?

**Comment by KevinMusgrave** [comment]
Yes, that would be the best way. I might as well add this to InferenceModel too.

**Comment by btseytlin** [comment]
@KevinMusgrave I will implement it and make a PR

**Comment by KevinMusgrave** [comment]
That would be great!

**Comment by btseytlin** [comment]
https://github.com/KevinMusgrave/pytorch-metric-learning/pull/134
